pkg/kubelet/metrics/metrics.go

```diff
@@ -40,13 +43,31 @@ const (
 	VolumeStatsInodesKey         = "volume_stats_inodes"
 	VolumeStatsInodesFreeKey     = "volume_stats_inodes_free"
 	VolumeStatsInodesUsedKey     = "volume_stats_inodes_used"
-	// Metrics keys of remote runtime operations
+	// Metric keys for remote runtime operations
 	RuntimeOperationsKey        = "runtime_operations"
 	RuntimeOperationsLatencyKey = "runtime_operations_latency_microseconds"
 	RuntimeOperationsErrorsKey  = "runtime_operations_errors"
-	// Metrics keys of device plugin operations
+	// Metric keys for device plugin operations
 	DevicePluginRegistrationCountKey = "device_plugin_registration_count"
 	DevicePluginAllocationLatencyKey = "device_plugin_alloc_latency_microseconds"
+	// Metric keys for node config sources
+	// AssignedNodeConfigSourceKey's metric reports the config source the Kubelet believes is currently assigned to it
+	AssignedNodeConfigSourceKey = "assigned_node_config_source"
```

Comment:
Why the `assigned_node` prefix? It's little big confusing, why not just `config_source` or `dynamic_config_source`?

Reply:
There are three config references you'd want the Kubelet to report:
1. What it believes its currently assigned source is.
2. What it believes its last-known-good source is.
3. Which of the above two it's actually using. 

`NodeConfigSource` is the type of the `ConfigSource` field on the `Node` object and is used to point the Kubelet at a config source, hence `node_config_source`.

Could do:
1. `node_config_source_assigned`
2. `node_config_source_last_known_good`
3. `node_config_source_using` (or some name other than `using`)